### PR TITLE
Get resolved module exports in symbol chain and not raw exports

### LINKED ISSUE #20657: Exported variable 'X' has or is using name 'Y' from external module "Z" but cannot be named.
**TypeScript Version:**  2.7.0-dev.20171212

**Code**

```ts
// thingA.ts
export interface ThingA { }

// thingB.ts
export interface ThingB { }

// things.ts
export { ThingA } from "./thingA";
export * from "./thingB";

// index.ts
import * as things from "./things";
export const thing1 = (param: things.ThingA) => null;
export const thing2 = (param: things.ThingB) => null;

// tsconfig.json
{
  "compilerOptions": {
    "declaration": true
  }
}
```

**Expected behavior:**

Project compiles without errors.

**Actual behavior:**

```
index.ts(3,14): error TS4023: Exported variable 'thing2' has or is using name 'ThingB' from external module "/Users/rob/Projects/tsctest/thingB" but cannot be named.
```

**Additional information:**

@weswigham fixed the following very similar issues in #19852:

* #5938
* #8612
* #19825

In fact, before his fix, line 2 of index.ts would also fail.

I've pushed this repro as a project here: https://github.com/robyoder/tsctest

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #20657

### CODE REVIEW

**Comment by robyoder** [comment]
That was fast! Thank you, @weswigham. ❤️

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -2216,7 +2216,8 @@ namespace ts {
 
                         // Look in the exported members, if we can find accessibleSymbolChain, symbol is accessible using this chain
                         // but only if the symbolFromSymbolTable can be qualified
-                        const accessibleSymbolsFromExports = resolvedImportedSymbol.exports ? getAccessibleSymbolChainFromSymbolTable(resolvedImportedSymbol.exports) : undefined;
+                        const candidateTable = getExportsOfSymbol(resolvedImportedSymbol);
+                        const accessibleSymbolsFromExports = candidateTable ? getAccessibleSymbolChainFromSymbolTable(candidateTable) : undefined;
```
Comment (andy-ms) [inline src/compiler/checker.ts]: Nit: Good place to use `&&` instead of `?:`

File: tests/baselines/reference/declarationEmitAliasExportStar.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+//// [tests/cases/compiler/declarationEmitAliasExportStar.ts] ////
+
+//// [thingA.ts]
+export interface ThingA { }
```
Comment (andy-ms) [inline tests/baselines/reference/declarationEmitAliasExportStar.js]: I can reproduce the error without `ThingA`:
**a.ts**
```ts
export interface I { }
```
**b.ts**
```ts
export * from "./a";
```
**c.ts**
```ts
import * as b from "./b";
export const f = (_: b.I) => {};
```

Comment (robyoder) [reply tests/baselines/reference/declarationEmitAliasExportStar.js]: Yep. I included ThingA in my issue report only to illustrate the difference between #19852 and #20657. No need to include it in the test.

**Comment by alan-agius4** [comment]
I downloaded `TypeScript`: `2.7.0-dev.20171221`  and the original problem is this there

```
index.ts(2,14): error TS4023: Exported variable 'thing1' has or is using name 'ThingA' from external module "C:/git/tsctest/thingA" but cannot be named.
index.ts(3,14): error TS4023: Exported variable 'thing2' has or is using name 'ThingB' from external module "C:/git/tsctest/thingB" but cannot be named.
```

**Pull request merged**